server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitorIT.java

```diff
@@ -78,5 +75,35 @@ public void testFloodStageExceeded() throws Exception {
                 equalTo("true")
             );
         });
+
+        final String newDataNodeName = internalCluster().startDataOnlyNode();
+        final String newDataNodeId = client().admin().cluster().prepareNodesInfo(newDataNodeName).get().getNodes().get(0).getNode().getId();
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareUpdateSettings(indexName)
+                .setSettings(
+                    Settings.builder()
+                        .put(INDEX_ROUTING_REQUIRE_GROUP_SETTING.getConcreteSettingForNamespace("_name").getKey(), newDataNodeName)
```

Comment:
I wonder if this test would not pass without this (I should totally try it out)...

Reply:
I see, yes, it might not always fail, we could move the shard to the new node and remove the block first. In practice it seems to fail here quite reliably but I will add another filter earlier in the test to make it always fail.

Reply:
I wonder if we should simply set this prior to starting the new node to ensure that the flood stage situation is not resolved already.

Also, I think we could validate that the index is no longer blocked at the end of this test?

Reply:
Well this is fun. Doing as you suggest found a case where we are missing a `reroute()`...

We need to know the name of the new node to apply this filter, but I reworked things not to need that in c76fc6b5af9fad323f706b45c8b302769a047fc1.